engine/table/src/main/java/io/deephaven/engine/table/impl/UpdateByBucketHelper.java

```diff
@@ -0,0 +1,336 @@
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.api.updateby.UpdateByControl;
+import io.deephaven.base.verify.Assert;
+import io.deephaven.chunk.*;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.*;
+import io.deephaven.engine.rowset.chunkattributes.OrderedRowKeys;
+import io.deephaven.engine.table.*;
+import io.deephaven.engine.table.impl.sources.ReinterpretUtils;
+import io.deephaven.engine.table.impl.ssa.LongSegmentedSortedArray;
+import io.deephaven.engine.table.impl.updateby.UpdateByWindow;
+import io.deephaven.util.datastructures.linked.IntrusiveDoublyLinkedNode;
+import org.apache.commons.lang3.mutable.MutableLong;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+import static io.deephaven.util.QueryConstants.NULL_LONG;
+
+/**
+ * An helper class of {@link UpdateBy} dedicated to zero key computation. This will manage the computation of a single
+ * bucket of rows.
+ */
+class UpdateByBucketHelper extends IntrusiveDoublyLinkedNode.Impl<UpdateByBucketHelper> {
+    protected final ColumnSource<?>[] inputSources;
+    // some columns will have multiple inputs, such as time-based and Weighted computations
+    final int[][] operatorInputSourceSlots;
+    final UpdateByOperator[] operators;
+    final UpdateByWindow[] windows;
+    final QueryTable source;
+    final UpdateBy.UpdateByRedirectionHelper redirHelper;
+    final UpdateByControl control;
+    final QueryTable result;
+
+    /** An array of {@link UpdateByWindow.UpdateByWindowContext}s for each window */
+    final UpdateByWindow.UpdateByWindowContext[] windowContexts;
+
+    /** store timestamp data in an SSA (if needed) */
+    final String timestampColumnName;
+    final LongSegmentedSortedArray timestampSsa;
+    final ColumnSource<?> timestampColumnSource;
+    final ModifiedColumnSet timestampColumnSet;
+
+    /** Indicates this bucket needs to be processed (at least one window and operator are dirty) */
+    boolean isDirty;
+
+    /**
+     * Perform an updateBy without any key columns.
```

Comment:
Seem incorrect

Reply:
This note is Incomplete.  This class handles a single bucket (either ZeroKey or already limited by key). Will update documentation.